### DISCUSSION TITLE
PG schema not considered in raw query

### DISCUSSION THREAD

**Discussion by TomasLulei** [sequelize/sequelize#15642] [Help & Questions] [answered]
Hello, I just run into issue similar to https://github.com/sequelize/sequelize/pull/13045. When I create connection into my database with specified schema, and then I run query (basically simple select column from table), the schema option is not considered and result in relation (table) does not exist.
`const client = new Sequelize(null, null, null, {username,password,dialect: 'postgres',database,host,port,pool,query: {raw: true}, schema: "my_schema"}); client.query(my_select()) .... error: relation "my_table" does not exist` 
"sequelize" version: 6.28.0
"pg" version": "8.9.0",
If I add schema directly to query -> select column from schema.table everything works, but I would like to use the schema option for client as we have several schemas, in multiple services.
Am I doing something wrong, do I need some additional setup?

**Comment by ephys** [comment, marked as answer]
We don't modify raw queries, they're used as-is, so if you don't specify a schema in your query it is going to use the [search path](https://www.postgresql.org/docs/current/ddl-schemas.html#DDL-SCHEMAS-PATH) to determine which table will be used

You can set the `searchPath` option when you call `sequelize.query`, be aware that `searchPath` in Sequelize is currently badly designed: the option will apply to all subsequent queries 😬
It also requires enabling this flag: https://github.com/sequelize/sequelize/issues/4771.

Still I highly recommend against using `searchPath` in its current state. My recommendation is to specify the schema in your query instead

**Reply by TomasLulei** [reply]
Thank you very much, that should actually work for us, as we do not work with multiple schemas in one query.
